internal/auth/ldap/repository_authenticate.go

```diff
@@ -108,11 +108,27 @@ func (r *Repository) Authenticate(ctx context.Context, authMethodId, loginName,
 	acct.Dn = authResult.UserDN
 
 	if authResult.UserAttributes != nil {
-		found, email := caseInsensitiveAttributeSearch(DefaultEmailAttribute, authResult.UserAttributes)
+		emailAttr := DefaultEmailAttribute
+		fullNameAttr := DefaultFullNameAttribute
+
+		attrMaps, err := am.convertAccountAttributeMaps(ctx)
+		if err == nil {
+			for _, attrMap := range attrMaps {
+				aam := attrMap.(*AccountAttributeMap)
+				if aam.ToAttribute == DefaultEmailAttribute {
+					emailAttr = aam.FromAttribute
+				}
+				if aam.ToAttribute == DefaultFullNameAttribute {
+					fullNameAttr = aam.FromAttribute
+				}
```

Comment:
nit: have a switch here, possibly even include a default that returns an error

Reply:
I don't think we should error if we get an unexpected attribute from the LDAP server - We should just ignore it, otherwise we could get into a situation where, for example, an LDAP setup for multiple downstream services could be incompatible with Boundary.

Reply:
I believe `convertAccountAttributeMaps` calls `ConvertToAccountToAttribute`
which will validate the attribs and error if there's an unexpected one.  

There's also a column constraint `to_attribute_valid_values` that validates the
valid values in the db.

Given these two constraints, it seems reasonable for us to have a switch here that
explicity returns an error before sending an invalid transaction to the db.